p2p/server.go

```diff
@@ -901,6 +980,426 @@ func (srv *Server) NodeInfo() *NodeInfo {
 	return info
 }
 
+func (srv *Server) loadKnownNodeInfos() {
+	fields := "ni.id, ni.node_id, nmi.hash, ip, tcp_port, remote_port, " +
+		"p2p_version, client_id, caps, listen_port, first_hello_at, last_hello_at, " +
+		"protocol_version, network_id, first_received_td, last_received_td, best_hash, genesis_hash, " +
+		"first_status_at, last_status_at, dao_fork"
+	maxIds := "SELECT node_id as nid, MAX(id) as max_id FROM node_info GROUP BY node_id"
+	nodeInfos := fmt.Sprintf("SELECT * FROM node_info x INNER JOIN (%s) max_ids ON x.id = max_ids.max_id", maxIds)
+	stmt := fmt.Sprintf("SELECT %s FROM (%s) ni INNER JOIN node_meta_info nmi ON ni.node_id=nmi.node_id", fields, nodeInfos)
+	rows, _ := srv.DB.Query(stmt)
+
+	type sqlObjects struct {
+		p2pVersion      sql.NullInt64
+		clientId        sql.NullString
+		caps            sql.NullString
+		listenPort      sql.NullInt64
+		firstHelloAt    sql.NullFloat64
+		lastHelloAt     sql.NullFloat64
+		protocolVersion sql.NullInt64
+		networkId       sql.NullInt64
+		firstReceivedTd sql.NullString
+		lastReceivedTd  sql.NullString
+		bestHash        sql.NullString
+		genesisHash     sql.NullString
+		daoForkSupport  sql.NullInt64
+		firstStatusAt   sql.NullFloat64
+		lastStatusAt    sql.NullFloat64
+	}
+
+	srv.KnownNodeInfos.Lock()
+	defer srv.KnownNodeInfos.Unlock()
+
+	for rows.Next() {
+		var (
+			rowid      uint64
+			nodeid     string
+			hash       string
+			ip         string
+			tcpPort    uint16
+			remotePort uint16
+			sqlObj     sqlObjects
+		)
+		err := rows.Scan(&rowid, &nodeid, &hash, &ip, &tcpPort, &remotePort,
+			&sqlObj.p2pVersion, &sqlObj.clientId, &sqlObj.caps, &sqlObj.listenPort,
+			&sqlObj.firstHelloAt, &sqlObj.lastHelloAt, &sqlObj.protocolVersion, &sqlObj.networkId,
+			&sqlObj.firstReceivedTd, &sqlObj.lastReceivedTd, &sqlObj.bestHash, &sqlObj.genesisHash,
+			&sqlObj.daoForkSupport, &sqlObj.firstStatusAt, &sqlObj.lastStatusAt)
```

Comment:
Just found out I screwed up the order here. This does not match the order used in line 987. `&sqlObj.daoForkSupport` should be the last column. I will push this change shortly.